process explorer/backend/ProcessExplorer/Processes/IProcessGenerator.cs

```diff
@@ -0,0 +1,247 @@
+﻿/* Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. */
+
+using ProcessExplorer.Entities;
+using System.Diagnostics;
+using System.Globalization;
+using System.Management;
+
+namespace LocalCollector.Processes
+{
+    public interface IProcessGenerator
+    {
+        public int? GetParentId(Process process);
+        public float GetMemoryUsage(Process process);
+        public float GetCPUUsage(Process process);
+        public List<ProcessInfoDto> GetChildProcesses(Process process);
+        public ProcessStartInfo KillProcessByName(string processName);
+        public ProcessStartInfo KillProcessById(int processId);
+        public void ProcessChanged(Process process);
+        public void WatchProcesses(List<int> processes);
+
+        public List<int> GetProcessIds(List<ProcessInfoDto> processes)
+        {
+            var list = new List<int>();
+            foreach (var process in processes)
+            {
+                if (process.PID != default)
+                    list.Add((int)process.PID);
+                if (process.Children != default)
+                {
+                    foreach (var child in process.Children)
+                    {
+                        if (child.PID != default)
+                            list.Add((int)child.PID);
+                    }
+                }
+            }
+            return list;
+        }
+    }
+
+    public class ProcessInfoLinux : IProcessGenerator
+    {
+        internal ProcessInfoLinux()
+        {
+
+        }
+        private string RunLinuxPSCommand(Process process, string command)
+        {
+            object locker = new object();
```

Comment:
If the intent here is that only one thread should be able to access the critical section below then this wouldn't work since each thread would create its own 'locker' instance.